x/stakeibc/keeper/community_pool.go

```diff
@@ -1,12 +1,104 @@
 package keeper
 
 import (
+	"errors"
+	"fmt"
+
+	sdkmath "cosmossdk.io/math"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/Stride-Labs/stride/v14/utils"
 )
 
+// For each hostZone with a valid community pool, trigger the ICQs and ICAs to transfer deposited tokens to holding address
+// Since ICQs and ICAs take time to complete, it is almost certain tokens swept in and processed will be swept out in a later epoch
+func (k Keeper) SweepInAllDepositedCommunityPoolTokens(ctx sdk.Context) error {
+	hostZones := k.GetAllActiveHostZone(ctx)
+	for _, hostZone := range hostZones {
+		if hostZone.CommunityPoolDepositIcaAddress != "" &&
+			hostZone.CommunityPoolHoldingAddress != "" &&
+			hostZone.CommunityPoolReturnIcaAddress != "" {
+				k.ICQCommunityPoolDepositICABalances(ctx, hostZone.ChainId)
+			}
+	}
+	return nil
+}
+
+// For each hostZone with a valid community pool, trigger the transfers of tokens in the holding address to the return ICA
+func (k Keeper) SweepOutAllReturningCommunityPoolTokens(ctx sdk.Context) error {
+	hostZones := k.GetAllActiveHostZone(ctx)
+	for _, hostZone := range hostZones {
+		if hostZone.CommunityPoolDepositIcaAddress != "" &&
+			hostZone.CommunityPoolHoldingAddress != "" &&
+			hostZone.CommunityPoolReturnIcaAddress != "" {
+				k.IBCReturnAllCommunityPoolTokens(ctx, hostZone)
+			}
+	}	
+	return nil
+}
+
+
+
+// Place holder function, will ICQ to see what balances exist in the deposit ICA on the community pool host zone
+// In the ICQ callback, for existing balances it will call IBCTransferCommunityPoolICATokensToStride with found token as input
+func (k Keeper) ICQCommunityPoolDepositICABalances(ctx sdk.Context, communityPoolHostZoneId string) error {
+	return nil
+}
+
+
 // ibc transfers tokens from the foreign hub community pool deposit ICA address onto Stride hub
-func (k Keeper) IBCTransferCommunityPoolICATokensToStride(ctx sdk.Context, token sdk.Coin) error {
-	k.Logger(ctx).Info("Transfering tokens from community pool to Stride...")
+// then as an atomic action, liquid stake the tokens with Autopilot commands in the ibc message memos
+func (k Keeper) IBCTransferCommunityPoolICATokensToStride(ctx sdk.Context, communityPoolHostZoneId string, token sdk.Coin) error {
+	k.Logger(ctx).Info(fmt.Sprintf("Transfering %d %s tokens from community pool deposit ICA to Stride hub holding address", token.Amount.Int64(), token.Denom))
+
+	// TODO: add safety check here on if the amount is greater than a threshold to avoid many, frequent, small transfers
+	//       threshold might be a config we tune to avoid ddos type attacks, for now using 0 as hard coded threshold
+	if token.Amount.LTE(sdkmath.ZeroInt()) {
+		k.Logger(ctx).Error(fmt.Sprintf("[IBCTransferCommunityPoolICATokensToStride] The amount %v to transfer did not meet the minimum threshold!", token.Amount))
+		return errors.New("Transfer Amount below threshold!")
```

Comment:
let's add a ticket to go through at the end and make new error types! also, as I mentioned on the call, feel free to remove these error logs and just log them from the upstream caller

Reply:
Yeah good point.  I am still going through and doing the errorsmod passing instead of logging at every level but it will take a bit.  Maybe I should do a small refactor with only fixing the error passing to log at the top level.